raft-tech/TANF-app
# Use hhs_id for ACF AMS logins

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1619

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.

1. Due to ACF AMS testing only available on the raft environment and requiring an AMS login via PIV/CAC card, only @ADPennington and myself will truly be able to test this.
1. However, we can verify that both Alex and my accounts now have the hhs_id values.

> *Demo GIF(s) and screenshots for testing procedure*
<details><summary>Click for gif</summary>

![1619](https://user-images.githubusercontent.com/84722778/163866077-336d952f-0af5-4901-8bcf-685148bb5715.gif)

</details>

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Verify `hhs_id` is in the AMS payload
- [x] Replace email as the unique identifier in the AMS login flow with `hhs_id`
- [x] Update or refactor other related sections of AMS login codebase as necessary
- [x] Update unit tests to account for this new identifier

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
